lib/ansible/modules/network/checkpoint/cp_mgmt_access_rule_facts.py

```diff
@@ -0,0 +1,242 @@
+    type: str
+  filter_settings:
+    description:
+      - Sets filter preferences.
+    type: list
+    suboptions:
+      search_mode:
+        description:
+          - When set to 'general', both the Full Text Search and Packet Search are enabled. In this mode, Packet Search will not match on 'Any'
+            object, a negated cell or a group-with-exclusion. When the search-mode is set to 'packet', by default, the match on 'Any' object, a negated cell
+            or a group-with-exclusion are enabled. packet-search-settings may be provided to change the default behavior.
+        type: str
+        choices: ['general', 'packet']
+      packet_search_settings:
+        description:
+          - When 'search-mode' is set to 'packet', this object allows to set the packet search preferences.
+        type: list
+        suboptions:
+          expand_group_members:
+            description:
+              - When true, if the search expression contains a UID or a name of a group object, results will include rules that match on at
+                least one member of the group.
+            type: bool
+          expand_group_with_exclusion_members:
+            description:
+              - When true, if the search expression contains a UID or a name of a group-with-exclusion object, results will include rules that
+                match at least one member of the "include" part and is not a member of the "except" part.
+            type: bool
+          match_on_any:
+            description:
+              - Whether to match on 'Any' object.
+            type: bool
+          match_on_group_with_exclusion:
+            description:
+              - Whether to match on a group-with-exclusion.
+            type: bool
+          match_on_negate:
+            description:
+              - Whether to match on a negated cell.
+            type: bool
+  limit:
+    description:
+      - No more than that many results will be returned.
+    type: int
+  offset:
+    description:
+      - Skip that many results before beginning to return them.
+    type: int
+  order:
+    description:
+      - Sorts results by the given field. By default the results are sorted in the ascending order by name.
+    type: list
+  package:
+    description:
+      - Name of the package.
+    type: str
+  use_object_dictionary:
+    description:
+      - N/A
+    type: bool
+  dereference_group_members:
+    description:
+      - Indicates whether to dereference "members" field by details level for every object in reply.
+    type: bool
+  show_membership:
+    description:
+      - Indicates whether to calculate and show "groups" field for every object in reply.
+    type: bool
+extends_documentation_fragment: checkpoint_facts
+"""
+
+EXAMPLES = """
+- name: show-access-rule
+  cp_mgmt_access_rule_facts:
+    hits_settings:
```

Comment:
same as above, `hits_settings` is defined as List but in example it's written as dict, if u want this way you'll have to define `type` as dict instead of `list`